skimage/filters/thresholding.py

```diff
@@ -580,9 +596,26 @@ def threshold_li(image, *, tolerance=None):
     image -= image_min
     tolerance = tolerance or np.min(np.diff(np.unique(image))) / 2
 
-    # Initial estimate
-    t_curr = np.mean(image)
-    t_next = t_curr + 2 * tolerance
+    # Initial estimate for iteration. See "initial_guess" in the parameter list
```

Comment:
Why one of these case would be decreased by image_min (due to line 596) and not the others? I would have expected that all of them should be reduced.

Reply:
@sciunto at this point, `image` is already in the new range (with the original `image_min` removed), so typically the mean/function should apply to the pixels in the new range. However, if the input is a float, it needs to be rescaled to the new range, because typically users will give the float value in the original data range.

it's important to note that the purpose here is to *increase* the image range in cases where it is negative, so `np.log` makes no sense.

Reply:
"image is already in the new range", You are correct; my remark was stupid.